platform/pulp/platform/models/repository.py

```diff
@@ -1,10 +1,342 @@
+
+    class Meta:
+        abstract = True
+        unique_together = ('name', 'repository')
+
+
+class RepositoryImporter(Importer):
+    """
+    A content importer that is associated with a repository..
+
+    Fields:
+
+    Relations:
+
+    :cvar repository: The associated repository.
+    :type repository: models.ForeignKey
+    """
+
+    repository = models.ForeignKey(
+        Repository, related_name='importers', on_delete=models.CASCADE)
+
+    class Meta:
+        unique_together = ('name', 'repository')
+
+
+class Distributor(Plugin):
+    """
+    An abstract content distributor.
+
+    Fields:
+
+    :cvar relative_url: The (relative) path component of the published url.
+    :type relative_url: models.TextField
+
+    :cvar last_published: When the last successful publish occurred.
+    :type last_published: models.DateTimeField
+
+    Relations:
+
+    """
+    relative_url = models.TextField(blank=True, default='')
+    last_published = models.DateTimeField(blank=True, null=True)
+
+    class Meta:
+        abstract = True
+
+
+class RepositoryDistributor(Distributor):
+    """
+    A content distributor that is associated with a repository.
+
+    Fields:
+
+    :cvar auto_publish: Indicates that the distributor may publish automatically
+        when the associated repository's content has changed.
+    :type auto_publish: models.BooleanField
+
+    Relations:
+
+    :cvar repository: The associated repository.
+    :type repository: models.ForeignKey
+
+    """
+    auto_publish = models.BooleanField(default=True)
+
+    repository = models.ForeignKey(
+        Repository, related_name='distributors', on_delete=models.CASCADE)
+
+    class Meta:
+        unique_together = ('name', 'repository')
+
+
+class GroupDistributor(Distributor):
+    """
+    A content distributor that is associated with a repository group.
+
+    Fields:
+
+    Relations:
+
+    :cvar group: The associated repository group.
+    :type group: models.ForeignKey
+
+    """
+    group = models.ForeignKey(
+        RepositoryGroup, related_name='distributors', on_delete=models.CASCADE)
+
+    class Meta:
+        unique_together = ('name', 'group')
+
+
+class RepositoryContent(Model):
+    """
+    Association between a repository and its contained content.
+
+    Fields:
+
+    :cvar created: When the association was created.
+    :type created: fields.DateTimeField
+
+    Relations:
+
+    :cvar content: The associated content.
+    :type content: models.ForeignKey
+
+    :cvar repository: The associated repository.
+    :type repository: models.ForeignKey
+    """
+    created = models.DateTimeField(auto_now_add=True)
+
+    content = models.ForeignKey('Content', on_delete=models.CASCADE)
+    repository = models.ForeignKey(Repository, on_delete=models.CASCADE)
+
+    def save(self, *args, **kwargs):
+        """
+        Save the association.
+        """
+        self.repository.last_content_added = datetime.utcnow()
```

Comment:
This (and anywhere else datetime.utcnow() is used) create a naive datetime object. Instead:
```python
from django.utils import timezone
timezone.now()
```

This generates a tz-aware datetime object in the default configured timezone which, in our case, is UTC.